### PR TITLE
prevent TPO updates if nothing changed

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/1616.

### CODE REVIEW

File: service/informer/informer.go [hunk 1]
```diff
@@ -166,7 +166,11 @@ func (s *Service) bootWithError() error {
 			Ref:  d.Sha,
 		}
 
-		TPO.Spec.Projects = ensureProject(TPO.Spec.Projects, newProject)
+		var updated bool
+		TPO.Spec.Projects, updated = ensureProject(TPO.Spec.Projects, newProject)
+		if !updated {
+			continue
+		}
```
Comment (xh3b4sd) [inline service/informer/informer.go:173]: This saves some writes to the Kubernetes API.

File: service/informer/informer.go [hunk 2]
```diff
@@ -229,25 +237,25 @@ func containsEmptyItems(projects []string) bool {
 	return false
 }
 
-func ensureProject(projects []draughtsmantprspec.Project, project draughtsmantprspec.Project) []draughtsmantprspec.Project {
+func ensureProject(projects []draughtsmantprspec.Project, project draughtsmantprspec.Project) ([]draughtsmantprspec.Project, bool) {
```
Comment (JosephSalisbury) [inline service/informer/informer.go]: I would appreciate some godoc here - when is true/false returned?

Comment (xh3b4sd) [reply service/informer/informer.go]: Added godoc.

**Comment by xh3b4sd** [comment]
Please have another look.

**Review by JosephSalisbury — APPROVED** [review]

**Pull request merged**